lib/Pool.js

```diff
@@ -456,29 +456,14 @@ class Pool {
   destroyAllNow() {
     this._log("force destroying all objects", "info");
 
-    const willDie = this._availableObjects.slice();
-    const todo = willDie.length;
-
     this._removeIdleScheduled = false;
     clearTimeout(this._removeIdleTimer);
 
-    return new Promise(resolve => {
-      if (todo === 0) {
-        return resolve();
-      }
-
-      let resource;
-      let done = 0;
-
-      while ((resource = willDie.shift())) {
-        this.destroy(resource.resource);
-        ++done;
-
-        if (done === todo && resolve) {
-          return resolve();
-        }
-      }
-    });
+    return Promise.all(
+      this._availableObjects.map(resource => {
+        return this.destroy(resource.resource);
+      })
+    ).then(() => {});
```

Comment:
Nice work! I was about to suggest using `Promise.map` instead to make this even more readable, but apparently Bluebird was removed (#2)...

Reply:
I was thinking about doing

```js
return this._availableObjects
  .reduce((promise, resource) => {
    return promise.then(() => this.destroy(resource.resource));
  }, Promise.resolve())
  .then(() => {});
```

instead of


```js
return Promise.all(
  this._availableObjects.map(resource => {
    return this.destroy(resource.resource);
  })
).then(() => {});
```

which would be the same as bluebirds `Promise.map` but I don't see any reason they need to be in series.

Reply:
I think that would actually be equivalent to `mapSeries`, not `map`, right? Regardless, the way you wrote it is great.